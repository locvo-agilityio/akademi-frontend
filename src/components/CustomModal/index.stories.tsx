import { BrowserRouter } from 'react-router-dom';
import { Meta, StoryObj } from '@storybook/react';
import { Text } from '@chakra-ui/react';

// Components
import CustomModal from '.';

const meta: Meta<typeof CustomModal> = {
  title: 'Components/CustomModal',
  component: CustomModal,
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    ),
  ],
};

export default meta;

type Story = StoryObj<typeof CustomModal>;

export const Primary: Story = {
  args: {
    title: 'Title',
    isOpen: true,
    body: <Text>Sample Modal</Text>,
  },
};
