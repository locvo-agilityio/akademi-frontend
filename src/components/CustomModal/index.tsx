import { ReactNode } from 'react';
import {
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalHeader,
  ModalOverlay,
} from '@chakra-ui/react';

interface TModalProps {
  maxW?: string;
  isOpen: boolean;
  title?: string;
  body?: ReactNode;
  hasCloseButton?: boolean;
  onClose: () => void;
}

const CustomModal = ({
  maxW = 'fit-content',
  isOpen,
  body,
  title = '',
  hasCloseButton = false,
  onClose,
}: TModalProps) => (
  <Modal isCentered isOpen={isOpen} onClose={onClose}>
    <ModalOverlay />
    <ModalContent minW={320} maxW={maxW}>
      <ModalHeader
        display="flex"
        justifyContent={hasCloseButton ? 'space-between' : 'center'}
        alignItems="center"
        w="full"
      >
        {title}
        {hasCloseButton && <ModalCloseButton position="unset" size="sm" />}
      </ModalHeader>
      {!!body && <ModalBody>{body}</ModalBody>}
    </ModalContent>
  </Modal>
);

export default CustomModal;
