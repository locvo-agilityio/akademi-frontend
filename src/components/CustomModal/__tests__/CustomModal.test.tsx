import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

// Components
import CustomModal from '..';

describe('CustomModal test cases', () => {
  const mockOnClose = jest.fn();

  const mockRenderBody = <></>;

  const setup = () =>
    render(
      <CustomModal
        isOpen={true}
        onClose={mockOnClose}
        body={mockRenderBody}
        title="mock title"
      />,
    );

  it('should render correctly', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('should render with default value', () => {
    const { container } = render(
      <CustomModal isOpen={true} onClose={mockOnClose} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should close CustomModal when close button is clicked', async () => {
    render(
      <CustomModal isOpen={true} onClose={mockOnClose} hasCloseButton={true} />,
    );
    await userEvent.click(screen.getByRole('button'));
    await waitFor(() => expect(mockOnClose).toHaveBeenCalled());
  });
});
