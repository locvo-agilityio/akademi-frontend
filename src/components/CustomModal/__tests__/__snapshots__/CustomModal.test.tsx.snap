// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomModal test cases should render correctly 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;

exports[`CustomModal test cases should render with default value 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;
